src/lib.rs

```diff
@@ -1,3 +1,66 @@
-#[test]
-fn it_works() {
+mod test;
+
+use std::collections::HashMap;
+
+/// A structure containing all the common attributes
+/// of the basic OSM data types
+/// 
+/// See: [OSM wiki - Common attributes](https://wiki.openstreetmap.org/wiki/Elements#Common_attributes)
+pub struct ElementInfo {
+    pub id: i64,
+    pub user: String,
+    pub uid: i32,
+    pub timestamp: String,
+    pub visible: bool,
+    pub version: i32,
+    pub changeset: i64
 }
+
+/// A node is a single Point holding `lat` and `lon` coordinates (WGS84 reference). 
+/// They are used to define standalone point features. They are also used for modelling 
+/// a Way or they can be a member of a relation.
+/// 
+/// See: [OSM wiki - Node](https://wiki.openstreetmap.org/wiki/Node)
+pub struct Node {
+    pub element_info: ElementInfo,
+    pub tags: HashMap<String, String>,
```

Comment:
Tags are also common between the different types. A HashMap takes a lot of space. As there is not so much tags in an OSM object, better to use a Vec or a flat_map::FlatMap